### DISCUSSION TITLE
[TypeScript] [select] 部分组件缺少props export

### DISCUSSION THREAD

**Discussion by galleonsZcc** [element-plus/element-plus#16910] [Feature Request]
Feature type: **`TypeScript`**

## Details

- Enhanced component name: `select`

## Description

部分组件虽然定义了props，但是没有使用ExtractPropTypes并export出去，例如el-select等。
我注意到el-input等组件有这些代码：
```typescript
export declare type InputProps = ExtractPropTypes<typeof inputProps>;
export declare const inputEmits: {};
export declare type InputEmits = typeof inputEmits;
export declare type InputInstance = InstanceType<typeof Input>;
```
但是el-select则没有，这就导致这部分组件在做扩展的时候有点困难，你们开发组件的时候并没有遵循统一的规范吗？

### Additional comments

_(empty)_

**Comment by warmthsea** [comment]
https://github.com/element-plus/element-plus/blob/b1bfb9fd1b52265ffd7737f4e9d81e0ac99c17d6/packages/components/select/src/token.ts#L45

如果你要使用  el-select 的 props 类型，你可以暂时使用这个
但是后面可能会统一，没保持一致可能是因为组件的写法有一定区别，el-select 这个组件的代码很多还是 vue3.2 出来之前写的，有一些历史遗留问题
